### PR TITLE
Allow custom options to be passed in WireCompiler

### PR DESCRIPTION
**Pull request by oldergod**

### CODE REVIEW

File: wire-compiler/src/main/java/com/squareup/wire/WireCompiler.kt [hunk 1]
```diff
@@ -372,6 +381,11 @@ class WireCompiler internal constructor(
             schemaHandlerFactoryClass = arg.substring(SCHEMA_HANDLER_FACTORY_CLASS_FLAG.length)
           }
 
+          arg.startsWith(CUSTOM_OPTION_FLAG) -> {
+            val (key, value) = arg.substring(SCHEMA_HANDLER_FACTORY_CLASS_FLAG.length).split(',')
+            customOptions[key] = value
```
Comment (oldergod) [inline wire-compiler/src/main/java/com/squareup/wire/WireCompiler.kt:386]: @JakeWharton made it simpler

Comment (JakeWharton) [reply wire-compiler/src/main/java/com/squareup/wire/WireCompiler.kt:386]: This is different that what you asked, though

File: wire-compiler/src/main/java/com/squareup/wire/WireCompiler.kt [hunk 2]
```diff
@@ -372,6 +381,11 @@ class WireCompiler internal constructor(
             schemaHandlerFactoryClass = arg.substring(SCHEMA_HANDLER_FACTORY_CLASS_FLAG.length)
           }
 
+          arg.startsWith(CUSTOM_OPTION_FLAG) -> {
+            val (key, value) = arg.substring(SCHEMA_HANDLER_FACTORY_CLASS_FLAG.length).split(',')
```
Comment (JakeWharton) [inline wire-compiler/src/main/java/com/squareup/wire/WireCompiler.kt:385]:
```suggestion
            val (key, value) = arg.substring(SCHEMA_HANDLER_FACTORY_CLASS_FLAG.length).split(',', limit = 2)
```

**Review by JakeWharton — APPROVED** [review]

File: wire-compiler/src/main/java/com/squareup/wire/WireCompiler.kt (hunk 1 above)
Comment (oldergod) [reply wire-compiler/src/main/java/com/squareup/wire/WireCompiler.kt:386]: Yep yep I made the API simpler. I was going with `options=key,value[,key,value]`

File: wire-compiler/src/main/java/com/squareup/wire/WireCompiler.kt (hunk 2 above)
Comment (oldergod) [reply wire-compiler/src/main/java/com/squareup/wire/WireCompiler.kt:385]: Nice

**Review by dnkoutso — APPROVED** [review]

**Pull request merged**